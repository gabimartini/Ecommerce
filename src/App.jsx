import React from 'react';
import { Route, Switch } from 'react-router-dom';
import './index.css';
import Layout from './containers/Layout';
import Home from './components/ProductSection/Home';
import Data from './components/ProductSection/Data';
import Basket from './components/Basket/Basket';
import Checkout from './components/Checkout/Checkout';
import Product from './components/ProductSection/Product';
import Orders from './components/Order/orders';
import Login from './components/Register/Login/Login'
import Register from './components/Register/register'

class App extends React.PureComponent {
  render() {
    return (
      <div>
        <Layout>
          <Switch>
            <Route path="/" exact component={Home} />
            <Route path="/Product/:id" component={Product} />
            <Route path="/All" component={Data} />
            <Route path="/Top" component={Data} />
            <Route path="/Bottom" component={Data} />
            <Route path="/Dress" component={Data} />
            <Route path="/Jumpsuit" component={Data} />
            <Route path="/Basket" component={Basket} />
            <Route path="/Checkout" component={Checkout} />
            <Route path="/Orders" component={Orders} />
            <Route path="/Register" component={Register} />
            <Route path="/Login" component={Login} />
          </Switch>
        </Layout>

      </div>
    );
  }
}

export default App;
