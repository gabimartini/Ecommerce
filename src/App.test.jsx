import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'
import App from './App';

//tirar o history das rotas e do app

test('App snapshot test', () => {
  const component = renderer.create(<BrowserRouter><App /></BrowserRouter>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
