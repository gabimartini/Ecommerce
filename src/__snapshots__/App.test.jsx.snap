// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<div>
  <header
    className="Toolbar"
  >
    <nav
      className="Main_Menu"
    >
      <ul>
        <a
          className="Home"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <div
          className="dropdown"
        >
          <li
            className="dropbtn"
          >
            Clothes
          </li>
          <div
            className="dropdowncontent"
          >
            <li
              className="dropdowncontentItem"
            >
              <a
                aria-current={null}
                className="NavLink"
                href="/All"
                onClick={[Function]}
              >
                All
              </a>
            </li>
            <li
              className="dropdowncontentItem"
            >
              <a
                aria-current={null}
                className="NavLink"
                href="/Top"
                onClick={[Function]}
              >
                Top
              </a>
            </li>
            <li
              className="dropdowncontentItem"
            >
              <a
                aria-current={null}
                className="NavLink"
                href="/Bottom"
                onClick={[Function]}
              >
                Bottom
              </a>
            </li>
            <li
              className="dropdowncontentItem"
            >
              <a
                aria-current={null}
                className="NavLink"
                href="/Jumpsuit"
                onClick={[Function]}
              >
                Jumpsuit
              </a>
            </li>
            <li
              className="dropdowncontentItem"
            >
              <a
                aria-current={null}
                className="NavLink"
                href="/Dress"
                onClick={[Function]}
              >
                Dress
              </a>
            </li>
          </div>
        </div>
      </ul>
    </nav>
    <a
      className="Basket"
      href="/Basket"
      onClick={[Function]}
    >
      <svg
        className="bi bi-bag"
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 5H2v9a1 1 0 0 0 1 1h10a1 1 0 0 0 1-1V5zM1 4v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V4H1z"
          fillRule="evenodd"
        />
        <path
          d="M8 1.5A2.5 2.5 0 0 0 5.5 4h-1a3.5 3.5 0 1 1 7 0h-1A2.5 2.5 0 0 0 8 1.5z"
        />
      </svg>
    </a>
  </header>
  <main>
    <h1>
      Home
    </h1>
  </main>
</div>
`;
